### PR TITLE
Bug 91742: Application error - no tabs

### PR DESCRIPTION
**Pull request by desousamario85**
**Changes to Web App**

- Added staticwebapp.config.json which helps with the routing and allows users to navigate directly to a page in the address bar. This also resolves the issue where in the bot framework , the tabs were showing 404 errors. 

**Updates in Bot Manifest File**

Added Additional Tabs: Feed, Users, Rewards, Wallet
The content URL for each tab needs to be updated to correct environment. currently this is hardcoded in the manifest files. 
Make sure you have added the web url as a Valid Domain in the file and the url has to be HTTPS, HTTP will not work.

**Deployment Pipelines**
Added new WriteEnv.js which will append the .env files with the LNBits variable in the env.dev file.
image
In the file we are only selecting selected variables, as the other variables related to MS Teams bot Framework

All environment variables have to be stored in GitHub Secrets, which will then set the variables during deployment.

### CODE REVIEW

**Comment by desousamario85** [comment]
Values are blank due to environment variables not being set.
Visual of the bot with the web app pages embedded.
image

**Comment by desousamario85** [comment]
Add variables in the workflow which will be replaced with the values saved in your Github sercets.
Each variable must be called the following names.
       REACT_APP_LNBITS_NODE_URL
       REACT_APP_LNBITS_USERNAME
       REACT_APP_LNBITS_PASSWORD
       REACT_APP_LNBITS_ADMINKEY
       REACT_APP_LNBITS_STORE_ID
       REACT_APP_TENANT_ID
       REACT_APP_AAD_CLIENT_ID

**Comment by BenGWeeks** [comment]
> Values are blank due to environment variables not being set. Visual of the bot with the web app pages embedded. image

Note the margin/padding needed @akash2017sky :-)

**Comment by BenGWeeks** [comment]
> Add variables in the workflow which will be replaced with the values saved in your Github sercets. Each variable must be called the following names. REACT_APP_LNBITS_NODE_URL REACT_APP_LNBITS_USERNAME REACT_APP_LNBITS_PASSWORD REACT_APP_LNBITS_ADMINKEY REACT_APP_LNBITS_STORE_ID REACT_APP_TENANT_ID REACT_APP_AAD_CLIENT_ID

I think I need a PR for the workflow with instructions for the secrets @akash2017sky? We should be deploying from this `main` please.

**Comment by BenGWeeks** [comment]
What is the Bug ID for this one please @EdiWeeks?

**Comment by EdiWeeks** [comment]
@BenGWeeks 
[Bug 91742](https://dev.azure.com/EvrosPowerPlatformTeam/Default/_workitems/edit/91742): Application error - no tabs

**Comment by BenGWeeks** [comment]
> Add variables in the workflow which will be replaced with the values saved in your Github sercets. Each variable must be called the following names. REACT_APP_LNBITS_NODE_URL REACT_APP_LNBITS_USERNAME REACT_APP_LNBITS_PASSWORD REACT_APP_LNBITS_ADMINKEY REACT_APP_LNBITS_STORE_ID REACT_APP_TENANT_ID REACT_APP_AAD_CLIENT_ID

Enhancement: Rename `REACT_APP_STORE_OWNER_EMAIL` to `REACT_APP_LNBITS_STORE_OWNER_EMAIL` to be consistent
Assigned to: @NataliaMilgevskaya 
FYI @EdiWeeks

**Comment by desousamario85** [comment]
variables are being populated, but a warning message is appearing in Teams.
image
@BenGWeeks FYI

**Comment by BenGWeeks** [comment]
Did a little look into this:

image

I also don't see the same tabs as you?

I assume the solutions should work when you are running locally too?

Check the secrets, as their are environment variables missing.

Also, make sure you are liaising with @akash2017sky on this as both working on the deployment flow atm.

**Comment by desousamario85** [comment]
@BenGWeeks make sure your manifest file for the bot is the latest.
It should contain the below.

image

**Comment by BenGWeeks** [comment]
Any changes would (should) be in the PR?

**Comment by desousamario85** [comment]
Code is in the PR. Screenshot is from the file changes listed.

**Comment by desousamario85** [comment]
**Bug 91800** MS Teams shows blank error banner has been opened, The issue requires additional time to resolve.

**Comment by BenGWeeks** [comment]
Do you think that pop-up is about cookies?

image

Alternatively, is it bubbling up the console.warnings?

image

image

Perhaps try changing console.warning in lnbitsServiceLocal to a console.log first.

File: appPackage/manifest.json [hunk 1]
```diff
@@ -75,9 +76,9 @@
                 "personalTab",
                 "channelTab"
             ], 
-            "name": "Home", 
-            "contentUrl": "http://localhost:3000/", 
-            "websiteUrl": "http://localhost:3000/"
+            "name": "Feed", 
+            "contentUrl": "https://ashy-pond-00441c81e.5.azurestaticapps.net/",
```
Comment (BenGWeeks) [inline appPackage/manifest.json:80]: We don't really want these hard coded, as Production will point to test, etc.

We want it to use `localhost:3000` / `<testurl>` / `<produrl>` etc.

**Comment by BenGWeeks** [comment]
OK, I think we see it on the homepage due to load times:

image

**Comment by BenGWeeks** [comment]
Will need to test pipeline after merge.

**Pull request merged**